client/post-editor/editor-sidebar/header.jsx

```diff
@@ -22,19 +24,32 @@ import DraftsButton from 'post-editor/drafts-button';
 import PostCountsData from 'components/data/post-counts-data';
 import QueryPostTypes from 'components/data/query-post-types';
 
+function goBack() {
+	if ( typeof window !== 'undefined' ) {
+		window.history.back();
+	}
+}
+
 function EditorSidebarHeader( { typeSlug, type, siteId, showDrafts, toggleDrafts, allPostsUrl, toggleSidebar } ) {
 	const isCustomPostType = ( 'post' !== typeSlug && 'page' !== typeSlug );
 	const className = classnames( 'editor-sidebar__header', {
 		'is-drafts-visible': showDrafts,
 		'is-loading': isCustomPostType && ! type
 	} );
-
-	let allPostsLabel;
-	switch ( typeSlug ) {
-		case 'post': allPostsLabel = translate( 'Posts' ); break;
-		case 'page': allPostsLabel = translate( 'Pages' ); break;
-		default: allPostsLabel = get( type, 'labels.menu_name' ) || translate( 'Loading…' );
+	const useBackButton = page.len > 0;
+	let closeLabel;
+	if ( useBackButton ) {
+		closeLabel = translate( 'Back' );
+	} else {
+		switch ( typeSlug ) {
+			case 'post': closeLabel = translate( 'Posts' ); break;
+			case 'page': closeLabel = translate( 'Pages' ); break;
+			default: closeLabel = get( type, 'labels.menu_name' ) || translate( 'Loading…' );
+		}
 	}
+	const closeButtonAction = useBackButton ? goBack : noop;
```

Comment:
Any reason not to use `page.back` ?

https://github.com/visionmedia/page.js/blob/aa60550/index.js#L205-L229